File: 487.max-consecutive-ones-ii/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        result = 0
        L = Z = 0
        N = len(nums)
        hasZero = False

        for R in range(N):
            num = nums[R]
            if num == 0:
                if hasZero:
                    result = max(result, R - L)
                    L = Z + 1
                    Z = R
                else:
                    Z = R
                    hasZero = True

        result = max(result, N - L)
        return result

File: 487.max-consecutive-ones-ii/solution.kt
import kotlin.math.*
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        // Keeps the last 1 position
        var L = 0
        // Keeps the last 0 position
        var Z = 0
        // Denotes we have zero between L and R
        var hasZero = false

        // answer
        var result = 0

        for (R in 0..nums.size - 1) {
            if (nums[R] == 0) {
                if (hasZero) {
                    result = max(result, R - L)
                    L = Z + 1
                    Z = R
                } else {
                    Z = R
                    hasZero = true
                }
            }
        }

        result = max(result, nums.size - L)
        return result
    }
}
